create_bags/bag_creator.py

```diff
@@ -1,18 +1,67 @@
-import logging
-
+import json
 from configparser import ConfigParser
+from subprocess import PIPE, Popen
+
 from .archivesspace import ArchivesSpaceClient
+from .helpers import create_tag, format_aspace_date, get_dates
 
 
-class BagCreator():
-    """docstring for BagCreator"""
+class BagCreator:
 
     def __init__(self):
-        logging.basicConfig(
-            datefmt='%m/%d/%Y %I:%M:%S %p',
-            filename='iiif_preparer.log',
-            format='%(asctime)s %(message)s',
-            level=logging.INFO)
         self.config = ConfigParser()
         self.config.read("local_settings.cfg")
-        self.ignore_list = self.config.get("IgnoreList", "ignore_list")
+        self.dest_location = self.config.get("Locations", "dest_location")
+        self.dart_command = self.config.get("DART", "dart")
+
+    def run(self, refid, rights_ids, files):
+        """
+        Args:
+        refid (str)
+        rights_ids (array)
+        """
+        # directory_to_bag = "some directory"
+        self.as_client = ArchivesSpaceClient(baseurl=self.config.get(
+            "ArchivesSpace", "baseurl"), username=self.config.get(
+            "ArchivesSpace", "username"), password=self.config.get(
+            "ArchivesSpace", "password"))
+        self.refid = refid
+        self.ao_uri = self.as_client.get_ao_uri(self.refid)
+        ao_data = self.as_client.get_ao_data(self.ao_uri)
+        dates = format_aspace_date(get_dates(ao_data))
+        self.job_params = self.construct_job_params(rights_ids, files, dates)
```

Comment:
Would consider expanding what gets returned from dates into two separate variables and then passing start and end date as separate arguments - it's a little weird to be passing a tuple that really is two different things, and having those as separate args will make the `construct_job_params` method a little clearer.

Reply:
Reupping this comment